nodes/stampzilla-server/metrics/metrics.go

```diff
@@ -17,38 +17,56 @@ type Logger interface {
 type Metrics struct {
 	previous map[string]interface{}
 	loggers  []Logger
-	queue    chan interface{}
+	queue    chan UpdatePackage
 }
 
+type UpdatePackage struct {
+	Node  *serverprotocol.Node
+	State map[string]interface{}
+}
+
+/* ----[ startup ]-------------------------------------------------*/
+
 func New() *Metrics {
 	return &Metrics{
 		make(map[string]interface{}),
 		nil,
-		make(chan interface{}, 100),
+		make(chan UpdatePackage, 100),
 	}
 }
 func (m *Metrics) AddLogger(l Logger) {
 	log.Infof("Adding logger: %T\n", l)
 	m.loggers = append(m.loggers, l)
 }
+
 func (m *Metrics) Start() {
 	go m.worker()
 }
+
 func (m *Metrics) worker() {
 	for s := range m.queue {
-		m.update(s)
+		m.update(s.Node, s.State)
 	}
 }
-func (m *Metrics) Update(s interface{}) {
-	m.queue <- s
+
+/* ----[ handle updates ]------------------------------------------*/
```

Comment:
Comments should follow go doc guidelines: http://blog.golang.org/godoc-documenting-go-code